### PR TITLE
BOR-4177 add data limit

### PR DESCRIPTION
**Pull request by RohitJ1995**

### CODE REVIEW

File: airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java [hunk 1]
```diff
@@ -233,7 +233,8 @@ private AutoCloseableIterator<AirbyteMessage> createReadIterator(final BigQueryD
         final AutoCloseableIterator<AirbyteMessage> iterator;
         final int sampleSize = (Integer) airbyteStream.getAdditionalProperties().get("sample_size");
         final int samplePercentage = (Integer) airbyteStream.getAdditionalProperties().get("sample_percentage");
-        iterator = getSampleDataStream(database, streamName, namespace, selectedDatabaseFields, table, emittedAt, sampleSize, samplePercentage);
+        Number maxByteSize = (Number) airbyteStream.getAdditionalProperties().get("max_data_size_mb");
```
Comment (void666) [inline airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java:236]: is there a default that we are adding, if this is not defined?

**Review by void666 — APPROVED** [review]
Let's add a default and merge.

**Review by jhecking — APPROVED** [review]

File: airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (RohitJ1995) [reply airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java]: done

File: airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java [hunk 2]
```diff
@@ -297,14 +298,31 @@ private AutoCloseableIterator<AirbyteMessage> getMessageIterator(
             final AutoCloseableIterator<JsonNode> recordIterator,
             final String streamName,
             final String namespace,
-            final long emittedAt) {
-        return AutoCloseableIterators.transform(recordIterator, r -> new AirbyteMessage()
+            final long emittedAt, Number maxDataSize) {
```
[the remark this answers is not in the document]
Comment (RohitJ1995) [reply airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java]: done

File: airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java [hunk 3]
```diff
@@ -297,14 +298,31 @@ private AutoCloseableIterator<AirbyteMessage> getMessageIterator(
...
+        { AirbyteMessage record = new AirbyteMessage()
                 .withType(AirbyteMessage.Type.RECORD)
                 .withRecord(new AirbyteRecordMessage()
                         .withStream(streamName)
                         .withNamespace(namespace)
                         .withEmittedAt(emittedAt)
-                        .withData(r)));
+                        .withData(r));
+            if (maxDataSize != null) {
+                if (totalByteSize[0] > (long) maxDataSize.intValue() * 1024 * 1024) {
```
[the remark this answers is not in the document]
Comment (RohitJ1995) [reply airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java]: done

File: airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java [hunk 4]
```diff
@@ -347,12 +347,13 @@ public static void main(final String[] args) throws Exception {
             new IntegrationRunner(source).run(args);
             LOGGER.info("completed source: {}", BorneoBigQuerySource.class);
         } catch (Exception e) {
-            LOGGER.error("Error in discover", e);
+            if (endExecuted && e instanceof IllegalStateException) {
+                LOGGER.warn("Ignoring IllegalStateException as endExecuted is true. e:", e);
+                return;
+            }
```
Comment (jhecking) [inline airbyte-integrations/connectors/source-borneo-bigquery/src/main/java/io/airbyte/integrations/source/borneobigquery/BorneoBigQuerySource.java:353]: Do you know what's causing the `IllegalStateException`?

**Pull request merged**